lib/facility.rb

```diff
@@ -1,14 +1,72 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility = {})
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def collect_fees(car)
+    if car.plate_type == :regular
+      @collected_fees += 100
+    elsif car.plate_type == :ev
+      @collected_fees += 200
+    elsif car.plate_type == :antique
+      @collected_fees += 25
+    end
+  end
+
+  def register_vehicle(car)
+    if @services.include?('Vehicle Registration')
+      car.registration_date = Date.today
+      car.plate_type
+      collect_fees(car)
+      @registered_vehicles.push(car)
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') == false
+      false
+    elsif registrant.age < 16 || registrant.permit? != true
+      false
+    else
+      registrant.license_data[:written] = true
+    end
+  end
+  
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') == false
+      false
+    elsif registrant.license_data[:written] != true
+      false
+    else
+      registrant.license_data[:license] = true
+    end
+  end
+  
+  def renew_drivers_license(registrant)
```

Comment:
Look at my comment for `administer_written_test`. I think this method could follow that same sort of logic flow.